src/components/SideBar/SideBar.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect, useRef } from 'react';
+import * as S from './Styles.SideBar';
+import Menu from './Menu/Menu';
+import SignIn from './SignIn/SignIn';
+
+const SideBar = () => {
+  const userMenu = useRef(null);
+  const [isOpen, setOpen] = useState(false);
+  const [showMode, setShowMode] = useState('menu');
+
+  const toggleMenu = e => {
+    e.stopPropagation();
+    setOpen(isOpen => !isOpen);
```

Comment:
```suggestion
    setOpen(prev => !prev);
```
boolean 값을 변경할 때는 이와 같은 방법으로 하는 것이 더 안전한 방법입니다!